app/src/test/java/com/example/memeitupapp/gridviewmodeltest/GridMemesViewModelTest.kt

```diff
@@ -0,0 +1,103 @@
+package com.example.memeitupapp.gridviewmodeltest
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Observer
+import com.example.memeitupapp.data.entity.Meme
+import com.example.memeitupapp.data.repository.MemeRepository
+import com.example.memeitupapp.ui.gridmemes.GridMemesViewModel
+import com.example.memeitupapp.util.Result
+import com.example.memeitupapp.util.Status
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.whenever
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.ObsoleteCoroutinesApi
+import kotlinx.coroutines.newSingleThreadContext
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.setMain
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNull
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.junit.MockitoJUnitRunner
+
+@RunWith(MockitoJUnitRunner::class)
+class GridMemesViewModelTest {
+
+    @ObsoleteCoroutinesApi
+    private var mainThreadSurrogate = newSingleThreadContext("UI thread")
+
+    @get:Rule
+    val instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    private lateinit var subject: GridMemesViewModel
+    private var memes: List<Meme> = mock()
+    private val mockedMemeRepository: MemeRepository = mock()
+
+    @ExperimentalCoroutinesApi
+    @ObsoleteCoroutinesApi
+    @Before
+    fun setup() {
+        Dispatchers.setMain(mainThreadSurrogate)
+
+        subject = GridMemesViewModel()
```

Comment:
here you should pass to the constructor of vm the mocked repo

Reply:
done, thanks